hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java

```diff
@@ -498,6 +498,14 @@ public boolean delete(Path f, boolean recursive) throws IOException {
     incrementCounter(Statistic.INVOCATION_DELETE, 1);
     statistics.incrementWriteOps(1);
     LOG.debug("Delete path {} - recursive {}", f, recursive);
+
+    String layOutVersion = adapter.getBucketLayoutVersion();
+    if (layOutVersion != null &&
+            OMConfigKeys.OZONE_OM_LAYOUT_VERSION_V1.equals(layOutVersion)) {
+      String key = pathToKey(f);
+      return adapter.deleteObject(key);
+    }
+
```

Comment:
For V1, we missed the root check delete at the client end.


For old buckets, this check is there.
```
      if (f.isRoot()) {
        LOG.warn("Cannot delete root directory.");
        return false;
      }
```

Reply:
Done in latest commit.